rfcs/0013-dns-publishing-strategy.md

```diff
@@ -0,0 +1,97 @@
+## Why
+Currently there is no way to instruct the Kuadrant Operator to publish or unpublish a DNS Record when certain conditions are met, this means that conditionally removing a DNS Record requires manual intervention and an understanding of the internal workings of the DNS-Operator.
+
+### Some Example Use Cases
+- DNS Failover (rapidly switch to alternative DNS Configuration) to a secondary site
+- Workload migration (removing workload from one cluster in favour of a new cluster)
+
+## What
+Add optional `unpublishRules` to the dns policy CRD, which will allow an administrator to define one or more rules which when satisfied will instruct the Kuadrant Operator to unpublish the records from the zone and see the results in a condition in the status of the policy.
+
+## How
+
+### Dependencies
+
+The [soft_delete](https://github.com/Kuadrant/dns-operator/issues/356) feature is required.
+
+### Diagrams
+
+#### Unpublish diagram
+![Diagram of DNS Publish strategy](./0013-dns-unpublishing-strategy-assets/unpublish-diagram.png)
+
+#### Health check changes diagram
+There are some changes to the health check probes also, below is a diagram showing how they will function once complete.
+
+![Health check updates](./0013-dns-unpublishing-strategy-assets/health-check-changes.png)
+
+https://miro.com/app/board/uXjVL32kOMY=/
+
+### CRD Updates
+#### DNS Record
+
+The DNS Record will no longer have a health check section in the spec.
+
+It will also have a new field in the spec `publish` if set to false the DNS Operator will attempt to lazily (i.e. not greedily) remove the DNS Records from the zone without interfering with other controllers records and without causing the potential for an NXDomain response. If set to true it will attempt to publish the endpoints to the zone as usual.
+
+#### DNS Policy
+The DNS Policy will have a new field added to their spec:
+
+```yaml
+unpublishRules:
+  <rule implemented using Common Expression Language>
+  <rule implemented using Common Expression Language>
+  <rule implemented using Common Expression Language>
+```
+### Kuadrant and DNS Operator operator changes
+
+### Health check probe creation
+The health checks are currently created by the DNS Operator, this will instead be moved to the Kuadrant-operator and the health checks will be owned by the DNS Policy that defined them, they will no longer have a failureThreshold as this can be defined in the unpublishRule, this section will no longer be copied to the DNS Record. They will continue to be reconciled by the DNS Operator.
```

Comment:
We can't remove the failureThreshold. That will be a breaking change to the API. In my opinion we could map failure theshold to a predefined cel predicate so as to maintain back compat but keep the impl the same for both

Reply:
By reconciled by the DNS Opertor I assume that means the health checks will still be performed by the DNS operator

Reply:
updated to apply these changes.